lib/osf-components/addon/components/editable-field/provider-metadata-manager/component.ts

```diff
@@ -0,0 +1,101 @@
+import { action } from '@ember/object';
+import { alias, and, reads } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency-decorators';
+import DS from 'ember-data';
+import Intl from 'ember-intl/services/intl';
+import Toast from 'ember-toastr/services/toast';
+
+import Registration, { ProviderMetadata } from 'ember-osf-web/models/registration';
+import CurrentUserService from 'ember-osf-web/services/current-user';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+
+export interface Args {
+    registration: Registration;
+}
+
+export interface ProviderMetadataManager {
+    save: () => void;
+    cancel: () => void;
+    startEditing: () => void;
+    inEditMode: boolean;
+    userCanEdit: boolean;
+    shouldShowField: () => boolean;
+    currentProviderMetadata: ProviderMetadata[];
+    isSaving: () => boolean;
+    fieldIsEmpty: () => boolean;
+    emptyFieldText: string;
+}
+
+function compareFieldValues(isEmpty: boolean, item: ProviderMetadata) {
+    return isEmpty && !item.field_value;
+}
+
+function deepCopy(providerMetadata: ProviderMetadata[]) {
+    return JSON.parse(JSON.stringify(providerMetadata));
+}
+
+export default class ProviderMetadataManagerComponent extends Component<Args> {
+    @service currentUser!: CurrentUserService;
+    @service intl!: Intl;
+    @service store!: DS.Store;
+    @service toast!: Toast;
+
+    @task({ withTestWaiter: true })
+    save = task(function *(this: ProviderMetadataManagerComponent) {
+        if (this.args.registration) {
+            try {
+                this.providerSpecificMetadata = deepCopy(this.currentProviderMetadata);
```

Comment:
It's a bit obscure/not intuitive the way `registration.providerSpecificMetadata` is updated  (via the alias) here because it requires knowing `alias` has two-way binding. Then we can get rid of the alias

```suggestion
                this.args.registration.providerSpecificMetadata = deepCopy(this.currentProviderMetadata);
```